nmdc_ms_metadata_gen/material_processing_generator.py

```diff
@@ -0,0 +1,349 @@
+import os
+import re
+import sys
+
+import nmdc_schema.nmdc as nmdc
+import pandas as pd
+from tqdm import tqdm
+
+from nmdc_ms_metadata_gen.changesheet_generator import (
+    ChangeSheetGenerator,
+    WorkflowSheetGenerator,
+    save_to_csv,
+)
+from nmdc_ms_metadata_gen.data_classes import ProcessGeneratorMap
+from nmdc_ms_metadata_gen.metadata_generator import NMDCWorkflowMetadataGenerator
+from nmdc_ms_metadata_gen.metadata_parser import YamlSpecifier
+from nmdc_ms_metadata_gen.study_metadata import MetadataSurveyor
+
+
+class MaterialProcessingMetadataGenerator(NMDCWorkflowMetadataGenerator):
+    """
+    This class provides functionality for generating material processing steps and processed samples based on an adjusted yaml outline and tracking the final outputs for changesheets.
+    """
+
+    def __init__(
+        self,
+        config_path: str,
+        output_path: str,
+        study_id: str,
+        yaml_outline_path: str,
+        sample_to_dg_mapping_path: str,
+        test: bool,
+    ):
+        super().__init__(
+            metadata_file="metadata_file",
```

Comment:
can we change these to NULL or empty strings?  Or can we make these parameters optional on the superclass?  I don't like passing these placeholder values to the superclass parameters.